src/main/java/com/bobocode/blyznytsia/bibernate/config/PersistenceConfiguration.java

```diff
@@ -0,0 +1,50 @@
+package com.bobocode.blyznytsia.bibernate.config;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+/**
+ * Record holds and provides access for basic persistence configuration parameters
+ * for one or more persistence units.
+ */
+public record PersistenceConfiguration(List<PersistenceUnitConfiguration> persistenceUnits) {
+
+  public PersistenceConfiguration(List<PersistenceUnitConfiguration> persistenceUnits) {
+    this.persistenceUnits = isNull(persistenceUnits) ? List.of() : persistenceUnits;
+  }
+
+  /**
+   * Record holds and provides access for parameters configuration of one persistence unit.
+   */
+  public record PersistenceUnitConfiguration(String name, DataSource dataSource) {
+
+    public PersistenceUnitConfiguration(String name, DataSource dataSource) {
+      this.name = requireNonNull(name, "Missed required property 'name' in configuration file");
+      this.dataSource =
+          requireNonNull(dataSource, "Missed required property 'dataSource' in configuration file");
+    }
+
+    /**
+     * Record holds and provides access for basic parameters that are mandatory for data source configuration.
+     */
+    public record DataSource(String jdbcUrl, String userName, String password, Integer poolSize,
+                             String driverClassName) {
+      public DataSource(String jdbcUrl, String userName, String password, Integer poolSize,
+                        String driverClassName) {
+        this.jdbcUrl =
+            requireNonNull(jdbcUrl, "Missed required property 'jdbcUrl' in configuration file");
```

Comment:
"Missed required property '%s' in configuration file" repeated 6 times. Perhaps, it makes sense to make a constant instead

Reply:
extracted to constant